pkg/api/types.go

```diff
@@ -495,6 +514,13 @@ type RawResources struct {
 	Memory *resource.Quantity `json:"memory"`
 }
 
+// Similar to RawResources, stores raw resource amounts. However, also stores the ID of the agent
+// notifying the VM of the granted resource request
+type SignedRawResources struct {
+	RawResources
+	Id AgentIdentification `json:"id"`
+}
```

Comment:
There's a few things I'd note here:

1. _In general_ the type names here serve to describe the _function_ of the message, rather than the _structure_. (e.g. `MoreResourcesRequest` or `DownscaleResult`) I think `RawResources` was kind of a unique exception. I'd initially though that the struct embedding was weird here, but I think it's just the name.
2. Simiarly, the comment above this type should have additional information about the function (i.e. "what is this used for? who sends it? why?")
3. Another thing I wanted to put on your radar: instead of writing `api.AgentMessage[api.AgentIdentification]` everywhere, we have a type alias `AgentIdentificationMessage`. A similar approach _may_ be useful here